backend/src/v2/driver/driver.go

```diff
@@ -285,7 +333,28 @@ func Container(ctx context.Context, opts Options, mlmd *metadata.Client, cacheCl
 		return execution, nil
 	}
 
-	execution.PodSpecPatch, err = makePodSpecPatch(opts.Container, opts.Component, executorInput, execution.ID, opts.PipelineName, opts.RunID)
+	// Write execution details to mlmd. This is usually done in launcher, but we skip launcher when acting Kubernetes specific actions.
+	// Thus we need to write the execution info in the driver.
+	if opts.Container.Image == "argostub/createpvc" {
+		executorOutput := &pipelinespec.ExecutorOutput{
+			ParameterValues: map[string]*structpb.Value{},
+			Artifacts:       map[string]*pipelinespec.ArtifactList{},
+		}
+		executorOutput.ParameterValues["name"] = structpb.NewStringValue(pvcName)
+
+		if err := mlmd.PublishExecution(ctx, createdExecution, executorOutput.GetParameterValues(), nil, pb.Execution_COMPLETE); err != nil {
+			return execution, fmt.Errorf("failed to publish execution: %w", err)
+		}
+		glog.Infof("Published this execution (createpvc).")
+	}
+	if opts.Container.Image == "argostub/deletepvc" {
+		if err := mlmd.PublishExecution(ctx, createdExecution, nil, nil, pb.Execution_COMPLETE); err != nil {
+			return execution, fmt.Errorf("failed to publish execution: %w", err)
+		}
+		glog.Infof("Published this execution (deletepvc).")
+	}
+
+	execution.PodSpecPatch, err = makePodSpecPatch(opts.Container, opts.Component, executorInput, execution.ID, opts.PipelineName, opts.RunID, opts.KubernetesConfig, opts.Task, dag, dagTasks)
```

Comment:
I don't have enough context here, but do you think we can skip the `makePodSpecPatch` for both `createpvc` and `deletepvc`?

Reply:
Good catch! I think we can skip it, since this patch is intended for the launcher container, which will not be created in this case. I also verified it manually.